src/main/java/ca/gc/aafc/collection/api/validation/GeoreferenceAssertionValidator.java

```diff
@@ -7,31 +7,37 @@
 import org.springframework.validation.Validator;
 
 import ca.gc.aafc.collection.api.entities.GeoreferenceAssertion;
+import ca.gc.aafc.collection.api.entities.GeoreferenceAssertion.GeoreferenceVerificationStatus;
 
 @Component
 public class GeoreferenceAssertionValidator implements Validator {
 
-    private MessageSource messageSource;
+  private MessageSource messageSource;
 
-    public GeoreferenceAssertionValidator(MessageSource messageSource) {
-        this.messageSource = messageSource;
-    }
-
-    @Override
-    public boolean supports(Class<?> clazz) {
-        return GeoreferenceAssertion.class.isAssignableFrom(clazz);
-    }
+  public GeoreferenceAssertionValidator(MessageSource messageSource) {
+    this.messageSource = messageSource;
+  }
 
-    @Override
-    public void validate(Object target, Errors errors) {
-        GeoreferenceAssertion georeferenceAssertion = (GeoreferenceAssertion) target;
+  @Override
+  public boolean supports(Class<?> clazz) {
+    return GeoreferenceAssertion.class.isAssignableFrom(clazz);
+  }
 
-        if (georeferenceAssertion.getCollectingEvent() != null && georeferenceAssertion.getCollectingEvent().getPrimaryGeoreferenceAssertion() == null) {
-            String errorMessage = messageSource.getMessage("georeferenceAssertion.collectingEvent.primaryGeoreferenceAssertion.null",
-                null, LocaleContextHolder.getLocale());
-            errors.rejectValue("collectingEvent", "georeferenceAssertion.collectingEvent.primaryGeoreferenceAssertion.null", errorMessage);
+  @Override
+  public void validate(Object target, Errors errors) {
```

Comment:
Overides and should carry the `@NonNull` ` public void validate(@NonNull Object target, @NonNull Errors errors)`